tests/v2/validation/provisioning/rke2/README.md

```diff
@@ -0,0 +1,202 @@
+```
+### Azure
+```json
+"azureCredentials": {
+   "clientId": "",
+   "clientSecret": "",
+     "subscriptionId": "",
+     "environment": "AzurePublicCloud"
+  },
+```
+### AWS
+```json
+"awsCredentials": {
+   "secretKey": "",
+   "accessKey": "",
+   "defaultRegion": ""
+  },
+```
+### Harvester
+```json
+"harvesterCredentials": {
+   "clusterId": "",
+   "clusterType": "",
+   "kubeconfigContent": ""
+},
+```
+### Google
+```json
+"googleCredentials": {
+    "authEncodedJson": ""
+}
+```
+
+## Machine RKE2 Config
+Machine RKE2 pool config is the last thing need for the RKE2 provisioning tests
+
+### AWS RKE2 Machine Config
+```json
+"awsMachineConfig": {
+    "region": "us-east-2",
+    "ami": "",
+    "instanceType": "t3a.medium",
+    "sshUser": "ubuntu",
+    "vpcId": "",
+    "volumeType": "gp2",
+    "zone": "a",
+    "retries": "5",
+    "rootSize": "16",
+    "securityGroup": ["rancher-nodes"]
+},
+```
+### Digital Ocean RKE2 Machine Config
+```json
+"doMachineConfig": {
+  "image": "ubuntu-20-04-x64",
+    "backups": false,
+    "ipv6": false,
+    "monitoring": false,
+    "privateNetworking": false,
+    "region": "nyc3",
+    "size": "s-2vcpu-4gb",
+    "sshKeyContents": "",
+    "sshKeyFingerprint": "",
+    "sshPort": "22",
+    "sshUser": "root",
+    "tags": "",
+    "userdata": ""
+},
+```
+### Linode RKE2 Machine Config
+```json
+"linodeMachineConfig": {
+  "authorizedUsers": "",
+  "createPrivateIp": false,
+  "dockerPort": "2376",
+  "image": "linode/ubuntu20.04",
+  "instanceType": "g6-standard-2",
+  "region": "us-west",
+  "rootPass": "",
+  "sshPort": "22",
+  "sshUser": "",
+  "stackscript": "",
+  "stackscriptData": "",
+  "swapSize": "512",
+  "tags": "",
+  "uaPrefix": ""
+},
+```
+### Azure RKE2 Machine Config
+```json
+"azureMachineConfig": {
+  "availabilitySet": "docker-machine",
+  "diskSize": "30",
+  "environment": "AzurePublicCloud",
+  "faultDomainCount": "3",
+  "image": "canonical:UbuntuServer:18.04-LTS:latest",
+  "location": "westus",
+  "managedDisks": false,
+  "noPublicIp": false,
+  "nsg": "",
+  "openPort": ["6443/tcp", "2379/tcp", "2380/tcp", "8472/udp", "4789/udp", "9796/tcp", "10256/tcp", "10250/tcp", "10251/tcp", "10252/tcp"],
+  "resourceGroup": "docker-machine",
+  "size": "Standard_D2_v2",
+  "sshUser": "docker-user",
+  "staticPublicIp": false,
+  "storageType": "Standard_LRS",
+  "subnet": "docker-machine",
+  "subnetPrefix": "192.168.0.0/16",
+  "updateDomainCount": "5",
+  "usePrivateIp": false,
+  "vnet": "docker-machine-vnet"
+},
+```
+### Harvester RKE2 Machine Config
+```json
+"harvesterMachineConfig": {
+  "diskSize": "40",
+  "cpuCount": "2",
+  "memorySize": "8",
+  "networkName": "default/ctw-network-1",
+  "imageName": "default/image-rpj98",
+  "vmNamespace": "default",
+  "sshUser": "ubuntu",
+  "diskBus": "virtio"
+}
+```
+
+## Custom Cluster
+For custom clusters, the below config is needed, only AWS/EC2 will work.
+**ensure you have nodeProviders in provisioningInput**
```

Comment:
Same as RKE1, this should be `Ensure you have nodeProviders in provisioningInput.`